packages/apps/slack/src/services/slack.ts

```diff
@@ -0,0 +1,166 @@
+import fs from "fs";
+import axios from "axios";
+import { Readable } from "stream";
+import { App, OAuthService, Organisation } from "@ocular/ocular";
+import {
+  IndexableDocument,
+  TransactionBaseService,
+  Logger,
+  AppNameDefinitions,
+  Section,
+} from "@ocular/types";
+import { ConfigModule } from "@ocular/ocular/src/types";
+import { DocType } from "@ocular/types/src/common";
+
+interface Config {
+  headers: {
+    Authorization: string;
+    Accept: string;
+  };
+}
+
+export default class SlackService extends TransactionBaseService {
+  protected oauthService_: OAuthService;
+  protected logger_: Logger;
+  protected container_: ConfigModule;
+
+  constructor(container) {
+    super(arguments[0]);
+    this.oauthService_ = container.oauthService;
+    this.logger_ = container.logger;
+    this.container_ = container;
+  }
+
+  async getSlackData(org: Organisation) {
+    return Readable.from(this.getSlackChannelsAndConversations(org));
+  }
+
+  async *getSlackChannelsAndConversations(
+    org: Organisation
+  ): AsyncGenerator<IndexableDocument[]> {
+    this.logger_.info(`Starting oculation of Slack for ${org.id} organisation`);
+
+    // Get Slack OAuth for the organisation
+    const oauth = await this.oauthService_.retrieve({
+      id: org.id,
+      app_name: AppNameDefinitions.SLACK,
+    });
+
+    if (!oauth) {
+      this.logger_.error(`No Slack OAuth found for ${org.id} organisation`);
+      return;
+    }
+
+    const config: Config = {
+      headers: {
+        Authorization: `Bearer ${oauth.token}`,
+        Accept: "application/json",
+      },
+    };
+
+    let documents: IndexableDocument[] = [];
+
+    try {
+      const slackChannels = await this.fetchSlackChannels(config)
+      for (const channel of slackChannels) {
+        const conversations = await this.fetchChannelConversations(channel.id,config);
+        for(const conversation of conversations){
+          const thread = await this.fetchThreadForConversation(channel.id,conversation.ts,config)
```

Comment:
This seems to be a bug, .ts does not exist on conversation. Similarly, it's non-existent for the references in this file. Consider changing to `conversation.id`

Reply:
Hey id will not work . Please refer to the following doc [https://api.slack.com/methods/conversations.history](https://api.slack.com/methods/conversations.history)

Reply:
Yes, refer to the other comment on this file, `ts` does not exist on the conversation object